client/app/services/services.js

```diff
@@ -401,3 +401,93 @@ angular.module('spkr.services', [])
   };
 
 })
+
+.factory('WordCloud', function() {
+
+  var layout;
+  var fill = d3.scale.category20();;
+
+  function wordMap(comments) {
+    // Takes an array of comments and maps to an array of objects with text and size
+    comments = comments.reduce(function(a,b) {
+      // Flatten the comments into one string
+      return a + ' ' + b;
+    })
+    var stopWords = /^(i|me|my|myself|we|us|our|ours|ourselves|you|your|yours|yourself|yourselves|he|him|his|himself|she|her|hers|herself|it|its|itself|they|them|their|theirs|themselves|what|which|who|whom|whose|this|that|these|those|am|is|are|was|were|be|been|being|have|has|had|having|do|does|did|doing|will|would|should|can|could|ought|i'm|you're|he's|she's|it's|we're|they're|i've|you've|we've|they've|i'd|you'd|he'd|she'd|we'd|they'd|i'll|you'll|he'll|she'll|we'll|they'll|isn't|aren't|wasn't|weren't|hasn't|haven't|hadn't|doesn't|don't|didn't|won't|wouldn't|shan't|shouldn't|can't|cannot|couldn't|mustn't|let's|that's|who's|what's|here's|there's|when's|where's|why's|how's|a|an|the|and|but|if|or|because|as|until|while|of|at|by|for|with|about|against|between|into|through|during|before|after|above|below|to|from|up|upon|down|in|out|on|off|over|under|again|further|then|once|here|there|when|where|why|how|all|any|both|each|few|more|most|other|some|such|no|nor|not|only|own|same|so|than|too|very|say|says|said|shall)$/
+    var punctuation = /[!"&()*+,-\.\/:;<=>?\[\\\]^`\{|\}~]+/g
+    var wordSeparators = /[\s\u3031-\u3035\u309b\u309c\u30a0\u30fc\uff70]+/g
+    var discard = /^(@|https?:)/
+    var maxLength = 30
```

Comment:
I don't think it really affects anything for our purposes. If a word is too long to fit on the word cloud (which is the real constraint), it will not be displayed anyway.